src/libraries/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -258,7 +259,7 @@ public void ExecuteAssemblyByName()
             }).Dispose();
         }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotMonoAOT))]
```

Comment:
Are all the tests that have this condition conditioned for the same reasoning? can the rationale be added to the PR description so that we can remember why these don't work for Mono AOT in the future?